crypto/evp/p_lib.c

```diff
@@ -1714,7 +1714,7 @@ void *evp_pkey_export_to_provider(EVP_PKEY *pk, OSSL_LIB_CTX *libctx,
         }
 
         /* Make sure that the keymgmt key type matches the legacy NID */
-        if (!ossl_assert(EVP_KEYMGMT_is_a(tmp_keymgmt, OBJ_nid2sn(pk->type))))
+        if (!EVP_KEYMGMT_is_a(tmp_keymgmt, OBJ_nid2sn(pk->type)))
```

Comment:
Any reason why you removed the assertion?

Reply:
Aha, was it possibly for the added negative test in test/keymgmt_internal/test.c?  Ok, I've no concerns then...